src/types/redis_zset.cc

```diff
@@ -142,7 +142,7 @@ rocksdb::Status ZSet::Card(const Slice &user_key, int *ret) {
   ZSetMetadata metadata(false);
   rocksdb::Status s = GetMetadata(ns_key, &metadata);
   if (!s.ok()) return s.IsNotFound() ? rocksdb::Status::OK() : s;
-  *ret = metadata.size;
+  *ret = static_cast<int>(metadata.size);
```

Comment:
Since the size of of type `uint32_t`, I'm wondering what will happen if `metadata.size` has a value more than INT32_MAX and we cast it here to `int` variable?
This question is related not only to this particular line but to many other across the codebase, and they also are present in this PR.

Reply:
I tried to keep the previous behavior. So, this cast was in previous version, but only it was implicit and now it is explicit.

> I'm wondering what will happen if metadata.size has a value more than INT32_MAX and we cast it here to int variable?

It is obvious that this can easily give rise to problems and bugs. But if you never met them before and I made all changes correctly then you shouldn't meet them in future (with a fairly high probability). In this particular case if `metadata.size` has a value more than `INT32_MAX` then I think we are talking about extremely large memory sizes (some gigabytes) and maybe it's unreal when we talk about meta information size.

But `UINT32_MAX` is greate than `INT32_MAX` only in 2 times. So, if `metadata.size` can be greater than `INT32_MAX` then it easily can be greater than `UINT32_MAX` and we should think about using 8 bytes integers (not 4 bytes) and it is not about this pr.